jax/_src/lax/utils.py

```diff
@@ -115,3 +116,12 @@ def standard_named_shape_rule(*avals, **kwargs):
 
 def _standard_weak_type_rule(*avals, **kwargs):
   return all(aval.weak_type for aval in avals)
+
+def _int_replaced_by_index(x):
+  try:
+    return operator.index(x)
+  except TypeError:
+    msg = ("using types support int(x) but don't support x.__index__ as int, "
+          "e.g. float, np.float64 will be deprecated")
```

Comment:
We should offer the user some clue here about the provenance of the error: imagine running a long & involved script and seeing several copies of this error, reported as coming from line 126 of `jax/_src/lax/utils.py`. What would you do to fix it?

Perhaps `_int_replaced_by_index` could accept the function name & variable name as strings, and put these in the warning to give context.

Alternatively, this may be an argument to inline this check in each place we use it, so that the warning is reported in the relevant context.